obfuscate.h

```diff
@@ -28,6 +28,9 @@ std::cout << obfuscated_string << std::endl;
 ----------------------------------------------------------------------------- */
 
 #pragma once
+#if __cplusplus >= 202002L
+	#include <type_traits>
```

Comment:
This introduces a dependency on the standard library, which is something that not all users may want or are able to do.

Since we don't get anything extra except for cleaner code, I think we should not go down this route and keep the library a zero dependency library.

Reply:
Didn't think about this, but you're right. Plus the code doesn't really get shorter/cleaner  since the fallback code is there anyway (and in the end it's probably implemented very similar or even identical).

Reply:
OK, I will remove it if you do not want to use the STL.